pyccel/ast/operators.py

```diff
@@ -490,14 +490,17 @@ class PyccelAdd(PyccelArithmeticOperator):
     """
     _precedence = 12
 
-    def __new__(cls, arg1, arg2):
+    def __new__(cls, arg1, arg2, simplify = False):
+        if simplify:
+            if isinstance(arg2, PyccelUnarySub):
+                return PyccelMinus(arg1, arg2.args[0], simplify = True)
```

Comment:
It would also be nice to simplify if both arguments are literals. I.e.:
```
if isinstance(arg1, Literal) and isinstance(arg2, Literal):
    dtype, precision = _calculate_dtype(arg1, arg2)
    if dtype is NativeReal():
        return LiteralFloat(arg1.python_value+arg2.python_value, precision=precision)
    elif ...:
        etc
```

Reply:
That way we can simplify `a = np.empty(10)` from the current print `allocate(a(0:10_C_LONG_LONG - 1_C_LONG_LONG))` to `allocate(a(0:9_C_LONG_LONG))`